templates/CRM/Contribute/Form/Contribution.tpl

```diff
@@ -488,7 +488,9 @@
      });
 
      function showHideCancelInfo(obj) {
-       if (obj.find(":selected").text() == 'Refunded' || obj.find(":selected").text() == 'Cancelled') {
+       var selected_status_id  = obj.find(":selected").val();
```

Comment:
I'm not sure what you mean about backward compatibility. This is the way to retrieve the value of a `select` field since jQuery 1.0. See http://api.jquery.com/val/